gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/domain_service/VerifyApiPathDomainServiceImpl.java

```diff
@@ -13,51 +13,49 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.gravitee.apim.infra.domain_service.api;
+package io.gravitee.apim.core.api.domain_service;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.base.Strings;
-import com.google.common.net.InternetDomainName;
-import io.gravitee.apim.core.api.domain_service.VerifyApiPathDomainService;
-import io.gravitee.apim.core.api.model.*;
+import io.gravitee.apim.core.api.model.Api;
+import io.gravitee.apim.core.api.model.ApiFieldFilter;
+import io.gravitee.apim.core.api.model.ApiSearchCriteria;
+import io.gravitee.apim.core.api.model.Path;
 import io.gravitee.apim.core.api.query_service.ApiQueryService;
 import io.gravitee.apim.core.environment.crud_service.EnvironmentCrudService;
+import io.gravitee.apim.core.exception.HttpListenerPathMissingException;
+import io.gravitee.apim.core.exception.InvalidHostException;
+import io.gravitee.apim.core.exception.PathAlreadyExistsException;
 import io.gravitee.definition.model.DefinitionVersion;
 import io.gravitee.definition.model.v4.listener.http.HttpListener;
-import io.gravitee.rest.api.service.v4.exception.HttpListenerPathMissingException;
-import io.gravitee.rest.api.service.v4.exception.InvalidHostException;
-import io.gravitee.rest.api.service.v4.exception.PathAlreadyExistsException;
-import java.io.IOException;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.context.annotation.Lazy;
-import org.springframework.stereotype.Service;
 
-@Service
 @Slf4j
 public class VerifyApiPathDomainServiceImpl implements VerifyApiPathDomainService {
```

Comment:
As we don't have any "infra" type, we could merge interface and implementation. I don't think we will have another type of implementation of this service 🤔